### PR TITLE
Feature/migration

### PR DESCRIPTION
**Pull request by DenisMisiuraItomy**

### CODE REVIEW

**Review by AndreyKomarichev10 — APPROVED** [review]

File: lib/flutter_call_kit.dart [hunk 1]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
           return null;
         }
         Map map = call.arguments.cast<String, dynamic>();
-        return _didReceiveStartCallAction(map["callUUID"], map["handle"]);
+        return _didReceiveStartCallAction!(map["callUUID"], map["handle"]);
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart:193]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 2]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
       case "onProviderReset":
         if (_onProviderReset == null) {
           return null;
         }
-        return _onProviderReset();
+        return _onProviderReset!();
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 3]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
       case "performAnswerCallAction":
         if (_performAnswerCallAction == null) {
           return null;
         }
-        return _performAnswerCallAction(
+        return _performAnswerCallAction!(
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 4]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
             call.arguments.cast<String, dynamic>()["callUUID"]);
       case "performEndCallAction":
         if (_performEndCallAction == null) {
           return null;
         }
-        return _performEndCallAction(
+        return _performEndCallAction!(
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 5]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
             call.arguments.cast<String, dynamic>()["callUUID"]);
       case "didActivateAudioSession":
         if (_didActivateAudioSession == null) {
           return null;
         }
-        return _didActivateAudioSession();
+        return _didActivateAudioSession!();
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 6]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
       case "didDeactivateAudioSession":
         if (_didDeactivateAudioSession == null) {
           return null;
         }
-        return _didDeactivateAudioSession();
+        return _didDeactivateAudioSession!();
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 7]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
       case "didDisplayIncomingCall":
         if (_didDisplayIncomingCall == null) {
           print("_didDisplayIncomingCall is null");
           return null;
         }
         Map map = call.arguments.cast<String, dynamic>();
-        return _didDisplayIncomingCall(map["error"], map["callUUID"],
+        return _didDisplayIncomingCall!(map["error"], map["callUUID"],
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 8]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
         Map map = call.arguments.cast<String, dynamic>();
-        return _didPerformSetMutedCallAction(map["muted"], map["callUUID"]);
+        return _didPerformSetMutedCallAction!(map["muted"], map["callUUID"]);
       case "didPerformDTMFAction":
         if (_didPerformDTMFAction == null) {
           return null;
         }
         Map map = call.arguments.cast<String, dynamic>();
-        return _didPerformDTMFAction(map["digits"], map["callUUID"]);
+        return _didPerformDTMFAction!(map["digits"], map["callUUID"]);
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart:240]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 9]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
-        return _didDisplayIncomingCall(map["error"], map["callUUID"],
+        return _didDisplayIncomingCall!(map["error"], map["callUUID"],
             map["handle"], map["localizedCallerName"], map["fromPushKit"]);
       case "didPerformSetMutedCallAction":
         if (_didPerformSetMutedCallAction == null) {
           return null;
         }
         Map map = call.arguments.cast<String, dynamic>();
-        return _didPerformSetMutedCallAction(map["muted"], map["callUUID"]);
+        return _didPerformSetMutedCallAction!(map["muted"], map["callUUID"]);
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 10]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
         Map map = call.arguments.cast<String, dynamic>();
-        return _didPerformDTMFAction(map["digits"], map["callUUID"]);
+        return _didPerformDTMFAction!(map["digits"], map["callUUID"]);
       case "didToggleHoldAction":
         if (_didToggleHoldAction == null) {
           return null;
         }
         Map map = call.arguments.cast<String, dynamic>();
-        return _didToggleHoldAction(map["hold"], map["callUUID"]);
+        return _didToggleHoldAction!(map["hold"], map["callUUID"]);
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 11]
```diff
@@ -190,66 +190,66 @@ class FlutterCallKit {
...
       case "handleStartCallNotification":
         if (_handleStartCallNotification == null) {
           return null;
         }
         Map map = call.arguments.cast<String, dynamic>();
-        return _handleStartCallNotification(map["handle"], map["video"]);
+        return _handleStartCallNotification!(map["handle"], map["video"]);
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy Please replace null assertion operator '!' with null aware operator '?'.

File: lib/flutter_call_kit.dart [hunk 12]
```diff
@@ -320,7 +320,11 @@ class FlutterCallKit {
   ///
   /// The [uuid] used for [startCall] or [displayIncomingCall]
   /// [reason] for the end call one of [EndReason]
-  Future<void> reportEndCallWithUUID(String uuid, EndReason reason) async {
+  Future<void> reportEndCallWithUUID(String? uuid, EndReason reason) async {
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy remove null aware operator '?' from 'uuid' argument

File: lib/flutter_call_kit.dart [hunk 13]
```diff
@@ -341,7 +345,10 @@ class FlutterCallKit {
   ///
   /// The [uuid] used for `startCall` or `displayIncomingCall`
 
-  Future<void> endCall(String uuid) async {
+  Future<void> endCall(String? uuid) async {
```
Comment (EgorLogachev) [inline lib/flutter_call_kit.dart]: @DenisMisiuraItomy remove null aware operator '?' from 'uuid' argument

**Pull request merged**